### PR TITLE
Lint yaml workflow update

### PR DESCRIPTION
**Pull request by Checksumz**
#### What type of PR is this:
/kind feature

### CODE REVIEW

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:16]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:15]: drop the `eye` emoji

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:24]: use git hashes instead of tags, they are mutable 

```
uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

apply this pattern to the other actions

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:30]:
```suggestion
          go-version: '1.22'
          check-latest: true 
```

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:26]: do you need to fetch the entire history?

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks for working on this
a few comments

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:41]: for this part you maybe can use this action that get all you need: https://github.com/tj-actions/changed-files

File: .github/workflows/yaml-lint-checker.yaml
Comment (cpanato) [inline .github/workflows/yaml-lint-checker.yaml:50]: this works? not 100% sure that this works

**Review by cpanato — CHANGES_REQUESTED** [review]
i think this should go to the release-actions repo we have and then re-use here, to avoid having code in this repo.

wdyt @saschagrunert @puerco @xmudrii ?

File: .github/scripts/validate_yaml.go
Comment (npolshakova) [reply .github/scripts/validate_yaml.go:46]: I think we want to print all the invalid files, and then exit after the loop.

File: .github/workflows/yaml-lint-checker.yaml
Comment (Checksumz) [reply .github/workflows/yaml-lint-checker.yaml:50]: Verified on my local branch that this allows to create a multiline variable for use in the next step.  Will send a screenshot with echo later

File: .github/scripts/validate_yaml.go
Comment (Checksumz) [reply .github/scripts/validate_yaml.go:46]: As Nina mentioned, this is to gather all the output into the variable on failure

**Comment by npolshakova** [comment]
> i think this should go to the release-actions repo we have and then re-use here, to avoid having code in this repo.
> 
> wdyt @saschagrunert @puerco @xmudrii ?

Are [these](https://github.com/kubernetes/release/tree/master/.github/workflows) the release actions workflows? We'd want to be able to run this on release notes PRs going into sig-release and block if there's any invalid yaml in the diff. That would prevent having to create a separate PR to unblock the release notes like this [one](https://github.com/kubernetes/sig-release/pull/2541).

File: .github/workflows/yaml-lint-checker.yaml
Comment (npolshakova) [reply .github/workflows/yaml-lint-checker.yaml:26]: As we edit the release notes, we usually keep the history: https://github.com/kubernetes/sig-release/pull/2542 
image

But then before we merge into the sig-release repo, we clean up everything into a single commit. 

I thought `fetch-depth: 1` only fetches the latest commit? So unless we get the entire history, the workflow will only run on a single commit (which might be fine since the last commit will have the entire diff, but makes it a little harder to identify the invalid yaml during the release notes review).

File: .github/workflows/yaml-lint-checker.yaml
Comment (cloudmelon) [reply .github/workflows/yaml-lint-checker.yaml:15]:
```suggestion
    name: Fixed Release Notes Check Yaml
```

File: .github/workflows/yaml-lint-checker.yaml
Comment (cloudmelon) [reply .github/workflows/yaml-lint-checker.yaml:50]: Release action from today's release meeting for reference : https://github.com/kubernetes-sigs/release-actions

File: .github/workflows/yaml-lint-checker.yaml
Comment (Checksumz) [reply .github/workflows/yaml-lint-checker.yaml:16]: cool done, do we normally just use latest than a specific version? Not clear because you mentioned using a specific commit on the

File: .github/workflows/yaml-lint-checker.yaml
Comment (Checksumz) [reply .github/workflows/yaml-lint-checker.yaml:24]: done

File: .github/workflows/yaml-lint-checker.yaml
Comment (Checksumz) [reply .github/workflows/yaml-lint-checker.yaml]: Done

File: .github/workflows/yaml-lint-checker.yaml
Comment (Checksumz) [reply .github/workflows/yaml-lint-checker.yaml:42]: checking this

**Comment by Checksumz** [comment]
@cpanato  i have implemented most of your suggestions apart from exit(0). 

@npolshakova  @cpanato  @saschagrunert  Just want to confirm if i need to close this PR and reopen it with https://github.com/kubernetes-sigs/release-actions and update it to be a reusable workflow ?

**Pull request closed without merging**
